pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -257,6 +258,18 @@ func (r *ReconcileOperConfig) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, nil
 	}
 
+	// Only allows live migration on SD managed clusters
+	inManagedCluster, err := util.IsRunningInManagedCluster(ctx, r.client)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+	if !inManagedCluster && operConfig.Spec.Migration != nil && operConfig.Spec.Migration.Mode == operv1.LiveNetworkMigrationMode {
+		r.status.SetDegraded(statusmanager.OperatorConfig, "MergeClusterConfig",
+			"invalid configuration: network type live migration is not supported on self-managed clusters. Use 'oc edit network.operator.openshift.io cluster' to fix.")
```

Comment:
does that mean the `Mode` field is actually mutable at runtime? there is some corner case here for mtu migration? (has(self.mode) && self.mode == 'Live'",message="networkType migration in mode other than 'Live' may not be configured at the same time as mtu migration")

Reply:
> does that mean the Mode field is actually mutable at runtime? 

yes

> there is some corner case here for mtu migration?

```
+openshift:validation:FeatureSetAwareXValidation:featureSet=CustomNoUpgrade;TechPreviewNoUpgrade,rule="!has(self.mtu) || !has(self.networkType) || self.networkType == '' || has(self.mode) && self.mode == 'Live'",message="networkType migration in mode other than 'Live' may not be configured at the same time as mtu migration"
type NetworkMigration struct {
```

Reply:
Anyway, this is outside the scope of this PR. @pliurh  can you confirm?

Reply:
might not be too much outside the scope of this PR because if you are giving a warning for a user to mutate it but you can't really mutate in all circumstances we must notify users of that limitation, but ok again to me if this is super urgent to get in I will not block for this.